src/Umbraco.Core/Services/IdkMap.cs

```diff
@@ -139,10 +144,12 @@ public void ClearCache(int id)
             try
             {
                 _locker.EnterWriteLock();
-                Guid key;
-                if (_id2Key.TryGetValue(id, out key) == false) return;
-                _id2Key.Remove(id);
-                _key2Id.Remove(key);
+                var match = _id2Key.Keys.SingleOrDefault(x => x.Id == id);
```

Comment:
You should be using _id2Key.TryGetValue . You already know the object type here so create a new Id2KeyCompositeKey and use that to find it. How come you don't want to use the object type here?